src/main/java/dev/j3fftw/litexpansion/items/PortableCharger.java

```diff
@@ -0,0 +1,224 @@
+import me.mrCookieSlime.Slimefun.cscorelib2.chat.ChatColors;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.scheduler.BukkitRunnable;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The {@link PortableCharger} is an item that opens
+ * a portable charging GUI that charges any
+ * {@link Rechargeable} item.
+ *
+ * @author FluffyBear
+ *
+ */
+public class PortableCharger extends SimpleSlimefunItem<ItemUseHandler> implements Listener, Rechargeable {
+
+    private final int[] border = { 5, 6, 7, 14, 16, 23, 24, 25 };
+
+    private final int powerSlot = 11;
+
+    private final int chargeSlot = 15;
+
+    private final int chargeSpeed = 20;
+
+    Plugin plugin = LiteXpansion.getInstance();
+
+    public PortableCharger() {
+        super(Items.LITEXPANSION, Items.PORTABLE_CHARGER, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            Items.REFINED_IRON, null, Items.REFINED_IRON,
+            null, Items.ADVANCED_CIRCUIT, null,
+            null, Items.ADVANCED_CIRCUIT, null
+        });
+
+        Bukkit.getPluginManager().registerEvents(this, LiteXpansion.getInstance());
+    }
+
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            e.cancel();
+
+            // Get variables
+            final Player p = e.getPlayer();
+            final ItemStack chargerItem = e.getItem();
+            final Rechargeable charger = (Rechargeable) SlimefunItem.getByItem(chargerItem);
+
+            // Create GUI Items
+            Inventory inventory = Bukkit.createInventory(null, 27, ChatColor.GOLD + "Portable Charger");
+
+            ItemStack backgroundItem = buildNonInteractable(Material.GRAY_STAINED_GLASS_PANE, null);
+            ItemStack borderItem = buildNonInteractable(Material.YELLOW_STAINED_GLASS_PANE, null);
+            ItemStack powerItem = buildNonInteractable(Material.GLOWSTONE, "&4Power");
+
+            // Build and open GUI
+            for (int i = 0; i < 27; i++)
+                inventory.setItem(i, backgroundItem);
+
+            for (int slot : border)
+                inventory.setItem(slot, borderItem);
+
+            inventory.setItem(powerSlot, powerItem);
+            updateSlot(inventory, powerSlot, "&6&lPower Remaining",
+                "&e" + charger.getItemCharge(chargerItem) + "J");
+            inventory.clear(chargeSlot);
+            p.openInventory(inventory);
+
+            // Task that triggers every second
+            new BukkitRunnable(){
+                public void run() {
```

Comment:
No need for this wasteful abstract class, I would suggest to rather use the lambda variant or to even consider making a new "PortableChargerTask" class, this would also improve readability for stacktraces or thread dumps as it would point to "PortableChargerTask:xy" rather than "PortableCharger:123456789~lambda:12345689".

Reply:
Whats the lambda variant?

Reply:
runTask(() -> ...);

Reply:
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/scheduler/BukkitScheduler.html

I dont think I can put an interval onto it?